static/js/components/SourceAlert.jsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+import { useForm, Controller } from "react-hook-form";
+
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+import FormControl from "@material-ui/core/FormControl";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormLabel from "@material-ui/core/FormLabel";
+import Radio from "@material-ui/core/Radio";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import MenuItem from "@material-ui/core/MenuItem";
+import Typography from "@material-ui/core/Typography";
+
+import { showNotification } from "baselayer/components/Notifications";
+import * as Actions from "../ducks/source";
+
+const useStyles = makeStyles((theme) => ({
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+  },
+  formContainer: {
+    display: "flex",
+    flexFlow: "column nowrap",
+    "& > div": {
+      margin: "0.5rem 0",
+    },
+  },
+}));
+
+const getFontStyles = (groupId, groupIds = [], theme) => ({
+  fontWeight:
+    groupIds.indexOf(groupId) === -1
+      ? theme.typography.fontWeightRegular
+      : theme.typography.fontWeightMedium,
+});
+
+const SourceAlert = ({ sourceId }) => {
+  const classes = useStyles();
+  const groups = useSelector((state) => state.groups.userAccessible);
+  const theme = useTheme();
+  const [selectedGroups, setSelectedGroups] = useState([]);
+  const { handleSubmit, getValues, reset, register, control } = useForm();
+  const dispatch = useDispatch();
+  const initialFormState = {
+    additionalNotes: "",
+    groupIds: [],
+    level: "soft",
+    sourceId,
+  };
+
+  const onSubmit = async () => {
+    const formData = {
+      ...initialFormState,
+      ...getValues(),
+    };
+    if (formData.groupIds.length === 0) {
+      dispatch(
+        showNotification("No target group(s) selected for alert", "error")
+      );
```

Comment:
Perhaps this would be better handled by RHF's validation functionality?

Reply:
e.g. https://github.com/skyportal/skyportal/blob/master/static/js/components/UploadPhotometry.jsx#L154